compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ConstantTypeChecker.java

```diff
@@ -1355,8 +1355,9 @@ private BType checkTupleType(BTupleType tupleType, BLangListConstructorExpr list
     }
 
     private BTupleType createNewTupleType(Location pos, List<BType> memberTypes, AnalyzerData data) {
-        BTypeSymbol tupleTypeSymbol = Symbols.createTypeSymbol(SymTag.TUPLE_TYPE, 0, Names.EMPTY,
-                data.env.enclPkg.symbol.pkgID, null, data.env.scope.owner, pos, SOURCE);
+        BTypeSymbol tupleTypeSymbol =
+                Symbols.createTypeSymbol(SymTag.TUPLE_TYPE, Flags.PUBLIC, Names.EMPTY,
```

Comment:
Why we set the flag to to always public, don't we need to infer from the `module-const-decl` node?

Reply:
Can you check behaviour of `createNewRecordType` method in the ConstantTypeChecker. Seems like there also we set the flag to 0. 

I checked this case `public const A = {a: "1", b: 2};` and I did not get the mentioned error.

Reply:
The reason we set the flag to public was because we didn't want to set flags in the TypeSymbol using the flags in the variable symbol. And the reason it is always public was that, if the variable was not public it would have been caught in earlier phases. 
In the `createNewRecordType` function, `BRecordTypeSymbol` that we pass is also created [here](https://github.com/ballerina-platform/ballerina-lang/blob/master/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ConstantTypeChecker.java#L2060) using the flags in the symbol of the variable rather than the type symbol itself. May be we need to address both of these places to have consistent behaviour. Either we set the flags using the flags in the symbol or do a check to see if the public flag is present and set the flag there. WDYT @prakanth97 @chiranSachintha ?

Reply:
As per the offline discussion we had, we decided to set the flag in the type symbol to `public` at this stage for both the occasions. We should ideally be able to remove the check [here](https://github.com/ballerina-platform/ballerina-lang/blob/master/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java#L3538). This [issue](https://github.com/ballerina-platform/ballerina-lang/issues/42300) is created to track the task.